src/meshapi/views/forms.py

```diff
@@ -352,13 +352,23 @@
     if nn_building.primary_node is not None:
         nn_install.node = nn_building.primary_node
     else:
-        try:
-            free_nn = get_next_available_network_number()
-        except ValueError as exception:
-            return Response({"detail": f"NN Request failed. {exception}"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        # Otherwise, try to use the install number if it is a valid number and unused
+        candidate_nn = nn_install.install_number
+        if (
+            candidate_nn < NETWORK_NUMBER_MIN
+            or candidate_nn > NETWORK_NUMBER_MAX
+            or len(Node.objects.filter(network_number=candidate_nn))
+        ):
+            # If that doesn't work, lookup the lowest available number and use that
+            try:
+                candidate_nn = get_next_available_network_number()
+            except ValueError as exception:
+                return Response(
+                    {"detail": f"NN Request failed. {exception}"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
```

Comment:
@james-otten I think this is in the same category as the other one we talked about, where this has a chance at leaking some information, but won't in this case because the `ValueError` is crafted specifically to be caught here

Reply:
Actually jk this is real. We cah just do `exception.args[0]` to fix it tho